table/tables/partition.go

```diff
@@ -166,6 +173,45 @@ func generatePartitionExpr(tblInfo *model.TableInfo) (*PartitionExpr, error) {
 	}, nil
 }
 
+func generateHashPartitionExpr(tblInfo *model.TableInfo) (*PartitionExpr, error) {
+	var column *expression.Column
+	// The caller should assure partition info is not nil.
+	pi := tblInfo.GetPartitionInfo()
+	ctx := mock.NewContext()
+	partitionPruneExprs := make([]expression.Expression, 0, len(pi.Definitions))
+	var buf bytes.Buffer
+	dbName := model.NewCIStr(ctx.GetSessionVars().CurrentDB)
+	columns := expression.ColumnInfos2ColumnsWithDBName(ctx, dbName, tblInfo.Name, tblInfo.Columns)
+	schema := expression.NewSchema(columns...)
+	for i := 0; i < int(pi.Num); i++ {
+		// If pi.Expr return negative number, this expression is not right, but this is only used to prune partition when do union now.
+		// Maybe mod(abs(-1),5) is better.
```

Comment:
@tiancaiamao PTAL

Reply:
? ok for currently or ` mod(abs(-1),5)` ?

Reply:
I mean, add some test cases, make sure the result is the same with MySQL.
Either `mod(abs(-1),5)` or `mod(-1,5)` looks good to me if the result is the same with MySQL.
Anyway, we can go over here if something is wrong.

Reply:
done.